src/Form/EntryForm.php

```diff
@@ -117,9 +130,21 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     //else {
     // TFA process is complete so finalize and authenticate user.
     //$context = $this->tfaManager->getContext($user);
-    //$tfa->finalize();
+
+    if (!empty($this->tfaLoginPlugins)) {
+      foreach ($this->tfaLoginPlugins as $plugin) {
+        if (method_exists($plugin, 'submitForm')) {
+          $plugin->submitForm($form, $form_state);
+        }
+      }
+    }
+
     user_login_finalize($user);
 
+    // TODO Should finalize() be after user_login_finalize or before?!
```

Comment:
I'm not sure, but let's try this and possibly note it in an issue to explore it moving fwd.